internal/cmd/skupper/listener/kube/listener_create.go

```diff
@@ -0,0 +1,217 @@
+		}
+
+		cmd.port, err = strconv.Atoi(args[1])
+		if err != nil {
+			validationErrors = append(validationErrors, fmt.Errorf("listener port is not valid: %s", err))
+		}
+		ok, err = numberValidator.Evaluate(cmd.port)
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("listener port is not valid: %s", err))
+		}
+	}
+
+	// Validate if there is already a listener with this name in the namespace
+	if cmd.name != "" {
+		listener, err := cmd.client.Listeners(cmd.namespace).Get(context.TODO(), cmd.name, metav1.GetOptions{})
+		if listener != nil && !errors.IsNotFound(err) && listener.Status.StatusMessage == "Ok" {
+			validationErrors = append(validationErrors, fmt.Errorf("there is already a listener %s created for namespace %s", cmd.name, cmd.namespace))
+		}
+	}
+
+	// Validate flags
+	if cmd.flags.routingKey != "" {
+		ok, err := resourceStringValidator.Evaluate(cmd.flags.routingKey)
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("routing key is not valid: %s", err))
+		}
+	}
+	if cmd.flags.host != "" {
+		//TBD what characters are not allowed
+	}
+
+	if cmd.flags.tlsSecret != "" {
+		// check that the secret exists
+		_, err := cmd.KubeClient.CoreV1().Secrets(cmd.namespace).Get(context.TODO(), cmd.flags.tlsSecret, metav1.GetOptions{})
+		if err != nil {
+			validationErrors = append(validationErrors, fmt.Errorf("tls-secret is not valid: does not exist"))
+		}
+	}
+	if cmd.flags.listenerType != "" {
+		ok, err := listenerTypeValidator.Evaluate(cmd.flags.listenerType)
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("listener type is not valid: %s", err))
+		}
+	}
+	if cmd.flags.output != "" {
+		ok, err := outputTypeValidator.Evaluate(cmd.flags.output)
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("output type is not valid: %s", err))
+		} else {
+			cmd.output = cmd.flags.output
+		}
+	}
+	return validationErrors
+}
+
+func (cmd *CmdListenerCreate) Run() error {
+
+	resource := v1alpha1.Listener{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "skupper.io/v1alpha1",
+			Kind:       "Listener",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cmd.name,
+			Namespace: cmd.namespace,
+		},
+		Spec: v1alpha1.ListenerSpec{
+			Host:           cmd.flags.host,
+			Port:           cmd.port,
+			RoutingKey:     cmd.flags.routingKey,
+			TlsCredentials: cmd.flags.tlsSecret,
+			Type:           cmd.flags.listenerType,
+		},
+	}
+
+	if cmd.output != "" {
+		encodedOutput, err := utils.Encode(cmd.output, resource)
+		fmt.Println(encodedOutput)
+		return err
+	} else {
+		_, err := cmd.client.Listeners(cmd.namespace).Create(context.TODO(), &resource, metav1.CreateOptions{})
+		return err
+	}
+}
+
+func (cmd *CmdListenerCreate) WaitUntilReady() error {
+	// the site resource was not created
+	if cmd.output != "" {
+		return nil
+	}
+
+	err := utils.NewSpinner("Waiting for create to complete...", 5, func() error {
```

Comment:
I have tried to create a listener a few seconds after deploying the skupper controller. It timed out at first, I suggest her to increase the number of tries (the minimum needed) in the Newspinner function for that use case.

Reply:
increasing to 10?  Or do we want it configurable like the timeout for creating a token?

Reply:
The latter sounds good, but I would start increasing to 10 tries